### PR TITLE
Add option to Add subreddit to Home Screen as shortcut

### PR DESCRIPTION
**Pull request by acolombo25**
As per [discussion #982](https://github.com/Docile-Alligator/Infinity-For-Reddit/discussions/982).

Here is a video recording demonstrating how it works:
https://github.com/Docile-Alligator/Infinity-For-Reddit/assets/9468299/31bce3ef-21e4-41b3-9ae2-583abd301383

### CODE REVIEW

**Comment by vishalnandagopal** [comment]
Thank you for implementing it!

File: app/src/main/java/ml/docilealligator/infinityforreddit/subreddit/shortcut/ShortcutManager.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+public class ShortcutManager {
+    private static ShortcutInfoCompat getInfo(Context context, String subreddit, Bitmap icon) {
+        final Intent shortcut = new Intent(context, ViewSubredditDetailActivity.class);
+        shortcut.setPackage(context.getPackageName());
+        shortcut.setAction(Intent.ACTION_MAIN);
+        shortcut.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+        shortcut.putExtra(ViewSubredditDetailActivity.EXTRA_SUBREDDIT_NAME_KEY, subreddit);
+
+        String shortcutId = BuildConfig.APPLICATION_ID + ".shortcut." + subreddit;
```
Comment (Docile-Alligator) [inline app/src/main/java/ml/docilealligator/infinityforreddit/subreddit/shortcut/ShortcutManager.java:22]: What if the user wants to add the same shortcut again? Does this make the previous one on the home screen disappear?

File: app/src/main/java/ml/docilealligator/infinityforreddit/activities/ViewSubredditDetailActivity.java [hunk 2]
```diff
@@ -1175,6 +1193,8 @@ public boolean onOptionsItemSelected(@NonNull MenuItem item) {
             intent.putExtra(SendPrivateMessageActivity.EXTRA_RECIPIENT_USERNAME, "r/" + subredditName);
             startActivity(intent);
             return true;
+        } else if (itemId == R.id.action_add_to_home_screen_view_subreddit_detail_activity) {
+            return ShortcutManager.requestPinShortcut(this, subredditName, subredditIconBitmap);
```
Comment (Docile-Alligator) [inline app/src/main/java/ml/docilealligator/infinityforreddit/activities/ViewSubredditDetailActivity.java:1197]: What if `subredditIconBitmap` is null?

File: app/src/main/java/ml/docilealligator/infinityforreddit/subreddit/shortcut/ShortcutManager.java (hunk 1 above)
Comment (acolombo25) [reply app/src/main/java/ml/docilealligator/infinityforreddit/subreddit/shortcut/ShortcutManager.java:22]: No, there will be two of the same shortcuts. I'm just using the system apis, there's no control on what's already on the home screen.

**Comment by Docile-Alligator** [comment]
Thank you!

**Pull request merged**